app/services/catalog/service_plan_json.rb

```diff
@@ -0,0 +1,37 @@
+module Catalog
+  class ServicePlanJson
+    include JsonSchemaReader
+
+    attr_reader :json
+
+    def initialize(opts)
+      if opts.key?(:service_plan_id)
+        @service_plans = ServicePlan.where(:id => opts[:service_plan_id])
+      elsif opts.key?(:portfolio_item_id)
+        @service_plans = ServicePlan.where(:portfolio_item_id => opts[:portfolio_item_id])
+      end
+
+      @schema = opts[:schema] || "modified"
+      @collection = opts[:collection]
+      @json = []
+    end
+
+    def process
+      @service_plans.each do |plan|
+        @reference = plan.portfolio_item.service_offering_ref
+        @portfolio_item_id = plan.portfolio_item.id
+        @service_plan = OpenStruct.new(
+          :id                 => plan.id,
+          :create_json_schema => plan.send(@schema),
+          :name               => plan.name,
+          :description        => plan.description
+        )
+
+        @json << read_json_schema("service_plan.erb")
+      end
+
+      @json = @json.first unless @collection
```

Comment:
Is the `@collection` instance variable supposed to be the switch for an `index` lookup vs a `show` lookup?

Reply:
Yep thats accurate - basically the only place we're going to use that is `GET /portfolio_items/:id/service_plans`. Everywhere else is going to be a show.

Reply:
I feel like we should change the logic to not iterate through all of the service plans in that case, then. If the collection is large, the two API calls will take the same amount of time even though one of them is only for a single object.

Reply:
Updated the logic to only iterate over the `relevant_service_plans` which goes off of the `:collection` option.